feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/presentation/governance/PushGovernanceSettingsViewModel.kt

```diff
@@ -0,0 +1,218 @@
+        }
+    }
+
+    fun referendaUpdatesClicked(item: PushGovernanceRVItem) {
+        _changedGovernanceSettingsList.updateValue {
+            it + item.model.copy(isReferendaUpdatesEnabled = !item.isReferendaUpdatesEnabled)
+                .disableCompletelyIfFeaturesDisabled()
+                .withKey()
+        }
+    }
+
+    fun delegateVotesClicked(item: PushGovernanceRVItem) {
+        _changedGovernanceSettingsList.updateValue {
+            it + item.model.copy(isDelegationVotesEnabled = !item.isDelegationVotesEnabled)
+                .disableCompletelyIfFeaturesDisabled()
+                .withKey()
+        }
+    }
+
+    fun tracksClicked(item: PushGovernanceRVItem) {
+        launch {
+            val selectedTracks = item.model.trackIds.mapToSet { it.value }
+            selectTracksRequester.openRequest(SelectTracksRequester.Request(item.chainId, selectedTracks, MIN_TRACKS))
+        }
+    }
+
+    fun clearClicked() {
+        _changedGovernanceSettingsList.value = emptyMap()
+    }
+
+    private fun mapCommunicatorModelToItem(
+        item: PushGovernanceSettingsPayload,
+        chain: Chain
+    ): PushGovernanceModel {
+        val tracks = item.tracksIds.toTrackIds()
+        return PushGovernanceModel(
+            chainId = item.chainId,
+            governance = item.governance,
+            chainName = chain.name,
+            chainIconUrl = chain.icon,
+            isEnabled = true,
+            isNewReferendaEnabled = item.newReferenda,
+            isReferendaUpdatesEnabled = item.referendaUpdates,
+            isDelegationVotesEnabled = item.delegateVotes,
+            trackIds = tracks
+        )
+    }
+
+    private fun mapItemToCommunicatorModel(item: PushGovernanceModel): PushGovernanceSettingsPayload {
+        return PushGovernanceSettingsPayload(
+            item.chainId,
+            item.governance,
+            item.isNewReferendaEnabled,
+            item.isReferendaUpdatesEnabled,
+            item.isDelegationVotesEnabled,
+            item.trackIds.fromTrackIds()
+        )
+    }
+
+    private fun PushGovernanceModel.disableCompletelyIfFeaturesDisabled(): PushGovernanceModel {
+        if (!isNewReferendaEnabled && !isReferendaUpdatesEnabled && !isDelegationVotesEnabled) {
+            return copy(isEnabled = false)
+        }
+
+        return this
+    }
+
+    private fun PushGovernanceModel.enableEverythingIfFeaturesDisabled(): PushGovernanceModel {
+        if (!isNewReferendaEnabled && !isReferendaUpdatesEnabled && !isDelegationVotesEnabled) {
+            return copy(isEnabled = true, isNewReferendaEnabled = true, isReferendaUpdatesEnabled = true, isDelegationVotesEnabled = true)
+        }
+
+        return this
+    }
+
+    private fun subscribeOnSelectTracks() {
+        selectTracksRequester.responseFlow
+            .onEach { response ->
+                val governanceVersion = Chain.Governance.V2 // Since we can use track selecting only for OpenGov
```

Comment:
I think it is better to not hardcode any specific assumptions here - imagine suppurting new governance types in the future. If we need governance verson here imgo it is better to return it in value of `responseFlow` as wel